### PR TITLE
Homepage PromoBlock structure

### PR DESCRIPTION
**Pull request by DamienBurkeWN**
### Testing

1. Not much to it, view Library/Structure/Promo Block stories and have a play. Compare with design mockup: https://www.figma.com/file/U2slrl7rJJHt69cKdfOWf2/Northants?node-id=3244%3A5671 (but note content text has been added)
2. Bonus bugfix 1: view Page Examples/News Article example/Example News Article and verify the blockquote no longer includes unprocessed markup
3. Bonus bugfix 2: AlertBanner no longer throws an error about window not existing at server build time, but this is best tested by pulling the library into your front end and so best as part of testing the frontend PR

### CODE REVIEW

File: src/library/pages/NewsArticle/NewsArticle.tsx [hunk 1]
```diff
@@ -40,20 +37,19 @@ export const NewsArticle: React.FC<NewsArticleProps> = ({ withImage, imagePlaceh
         <p>In Northamptonshire £750,000 from the grant is being used to provide shopping vouchers for families with children who are eligible for Free School Meals with vouchers being distributed to the families by their schools.</p>
 
         <p>Cllr Fiona Baker, county council cabinet member for Children, Families and Education, said:</p>
-        <BlockQuote quote="<p>“I’m delighted that we’re able to continue this support for the families and pupils who are in most need of our help.</p>
+
+        <BlockQuote quote={<><p>“I’m delighted that we’re able to continue this support for the families and pupils who are in most need of our help.</p>
```
Comment (DamienBurkeWN) [inline src/library/pages/NewsArticle/NewsArticle.tsx:41]: Bonus bugfix

File: src/library/structure/AlertBanner/AlertBanner.tsx [hunk 2]
```diff
@@ -40,6 +40,7 @@ function useLocalStorage(key, initialValue) {
     // State to store our value
     // Pass initial state function to useState so logic is only executed once
     const [storedValue, setStoredValue] = useState(() => {
+      if (typeof window === 'undefined') return initialValue; // e.g. at server-side build time
```
Comment (DamienBurkeWN) [inline src/library/structure/AlertBanner/AlertBanner.tsx:43]: Bonus bugfix no 2!

File: src/library/structure/PromoBlock/PromoBlock.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                                partialVisibility: true
+                            }}
+                        >
+                            {src => 
+                                <Styles.PromoImage background={src} role="img" aria-label={promo.imageAltText ? promo.imageAltText : ""} />
+                            }
+                        </LazyImage>
+                        <Styles.PromoText>
+                            <Styles.PromoHeadline level={3} text={promo.title} />
+                            <Styles.PromoContent dangerouslySetInnerHTML={{__html: DOMPurify.sanitize(promo.content)}} />
```
Comment (DamienBurkeWN) [inline src/library/structure/PromoBlock/PromoBlock.tsx:31]: XSS risk, whilst low as we're only supplying content entered in our own CMS, further mitigated by use of DOMPurify: https://github.com/cure53/DOMPurify

File: src/library/structure/PageMain/PageMain.styles.js [hunk 4]
```diff
@@ -72,7 +72,7 @@ export const Container = styled.main`
     // TABLE STYLES
     .table-container {
         max-width: 100%;
-        overflow-y: scroll;
+        overflow-y: auto;
```
Comment (DamienBurkeWN) [inline src/library/structure/PageMain/PageMain.styles.js:75]: Squeezing this in: https://github.com/FutureNorthants/northants-website/issues/58

**Review by chrisrhymes — APPROVED** [review]
Looks good. There just seems to be a mix of 2 and 4 spaces indentation across different files.   Could be worth adding editor config and prettierrc files to this repo to match the frontend formatting.

File: src/library/structure/PromoBlock/PromoBlock.styles.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+    display: flex;
+    flex-wrap: wrap;
+  }
+`;
+
+/**
+ * A promo tile, containing image and text area
+ */
+export const PromoTile = styled.a`
+  background: ${props => props.theme.cardinal_name === "north" ? props.theme.theme_vars.colours.white : props.theme.theme_vars.colours.grey_light} !important;
```
[the remark this answers is not in the document]
Comment (DamienBurkeWN) [reply src/library/structure/PromoBlock/PromoBlock.styles.js:26]: Following the pattern elsewhere in the design system unfortunately, see news articles block for instance. Styled components really don't make it easy to visualise the overall styling hierarchy that ends up being in use and there are some really unpredictable overrides of what you expect.

**Comment by AnthonyTaylor** [comment]
All look good to me

**Pull request merged**